bokeh/models/annotations.py

```diff
@@ -111,8 +112,9 @@ class BoxAnnotation(Annotation):
 
     """
 
-    left = Either(Auto, NumberSpec(), default=None, help="""
-    The x-coordinates of the left edge of the box annotation.
+    left = NumberSpec(default=None, help="""
+    The x-coordinates of the left edge of the box annotation. If value is set
+    to `None`, the box annotation will continue to the edge of the plot area.
     """)
```

Comment:
This will break code. @mattpap added a transform option of some sort, that could be used to turn "auto" into `None` which might be one option.